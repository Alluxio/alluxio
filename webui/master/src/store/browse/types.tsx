/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {AxiosResponse} from 'axios';

import {IFileBlockInfo, IFileInfo, ICommonState} from '@alluxio/common-ui/src/constants';
import {IInit} from "../init/types";

export interface IBrowse {
  'accessControlException': string;
  'blockSizeBytes': string;
  'currentDirectory': IFileInfo;
  'currentPath': string;
  'debug': boolean;
  'fatalError': string;
  'fileBlocks': IFileBlockInfo[],
  'fileData': string,
  'fileDoesNotExistException': string;
  'fileInfos': IFileInfo[];
  'highestTierAlias': string,
  'invalidPathError': string;
  'invalidPathException': string;
  'masterNodeAddress': string;
  'ntotalFile': number;
  'pathInfos': IFileInfo[];
  'showPermissions': boolean;
  'viewingOffset': number;
}

export enum BrowseActionTypes {
  FETCH_REQUEST = '@@browse/FETCH_REQUEST',
  FETCH_SUCCESS = '@@browse/FETCH_SUCCESS',
  FETCH_ERROR = '@@browse/FETCH_ERROR'
}

export interface IBrowseState {
  readonly data: IBrowse;
  readonly errors?: AxiosResponse;
  readonly loading: boolean;
  readonly response?: AxiosResponse;
}
