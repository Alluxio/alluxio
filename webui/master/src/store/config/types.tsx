/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {AxiosResponse} from 'axios';

import {IConfigTriple} from '../../constants';
import {ICommonState} from "@alluxio/common-ui/src/constants";

export interface IConfig {
  configuration: IConfigTriple[];
  whitelist: string[];
}

export enum ConfigActionTypes {
  FETCH_REQUEST = '@@config/FETCH_REQUEST',
  FETCH_SUCCESS = '@@config/FETCH_SUCCESS',
  FETCH_ERROR = '@@config/FETCH_ERROR'
}

export interface IConfigState {
  readonly data: IConfig;
  readonly loading: boolean;
  readonly errors?: AxiosResponse;
  readonly response?: AxiosResponse;
}
