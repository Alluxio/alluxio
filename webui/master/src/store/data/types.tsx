/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {AxiosResponse} from 'axios';

import {IFileInfo, ICommonState} from '@alluxio/common-ui/src/constants';

export interface IData {
  'fatalError': string;
  'fileInfos': IFileInfo[],
  'inAlluxioFileNum': number;
  'masterNodeAddress': string;
  'permissionError': string;
  'showPermissions': boolean;
}

export enum DataActionTypes {
  FETCH_REQUEST = '@@data/FETCH_REQUEST',
  FETCH_SUCCESS = '@@data/FETCH_SUCCESS',
  FETCH_ERROR = '@@data/FETCH_ERROR'
}

export interface IDataState {
  readonly data: IData;
  readonly loading: boolean;
  readonly errors?: AxiosResponse;
  readonly response?: AxiosResponse;
}