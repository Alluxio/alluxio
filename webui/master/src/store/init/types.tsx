/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {AxiosResponse} from 'axios';

import {IFileInfo, ICommonState} from '@alluxio/common-ui/src/constants';

export interface IInit {
  'debug': boolean;
  'newerVersionAvailable': boolean;
  'proxyDownloadFileApiUrl': {
    'prefix': string;
    'suffix': string;
  };
  'refreshInterval': number;
  'securityAuthorizationPermissionEnabled': boolean;
  'webFileInfoEnabled': boolean;
  'workerPort': number;
}

export enum InitActionTypes {
  FETCH_REQUEST = '@@init/FETCH_REQUEST',
  FETCH_SUCCESS = '@@init/FETCH_SUCCESS',
  FETCH_ERROR = '@@init/FETCH_ERROR'
}

export interface IInitState {
  readonly data: IInit;
  readonly errors?: AxiosResponse;
  readonly loading: boolean;
  readonly response?: AxiosResponse;
}
