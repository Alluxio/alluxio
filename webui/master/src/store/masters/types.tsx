/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import { AxiosResponse } from 'axios';

import { IMasterInfo } from '../../constants';

export interface IMasters {
  debug: boolean;
  lostMasterInfos: IMasterInfo[];
  standbyMasterInfos: IMasterInfo[];
  primaryMasterInfo: IMasterInfo;
}

export enum MastersActionTypes {
  FETCH_REQUEST = '@@masters/FETCH_REQUEST',
  FETCH_SUCCESS = '@@masters/FETCH_SUCCESS',
  FETCH_ERROR = '@@masters/FETCH_ERROR',
}

export interface IMastersState {
  readonly data: IMasters;
  readonly errors?: AxiosResponse;
  readonly loading: boolean;
  readonly response?: AxiosResponse;
}
