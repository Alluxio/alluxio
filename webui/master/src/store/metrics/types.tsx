/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {LineSerieData} from '@nivo/line';
import {AxiosResponse} from 'axios';

import {ICounter} from '@alluxio/common-ui/src/constants';

export interface IMetrics {
  cacheHitLocal: string;
  cacheHitRemote: string;
  cacheMiss: string;
  masterCapacityFreePercentage: number;
  masterCapacityUsedPercentage: number;
  masterUnderfsCapacityFreePercentage: number;
  masterUnderfsCapacityUsedPercentage: number;
  rpcInvocationMetrics: {
    [key: string]: ICounter;
  };
  timeSeriesMetrics: LineSerieData[];
  totalBytesReadLocal: string;
  totalBytesReadLocalThroughput: string;
  totalBytesReadDomainSocket: string;
  totalBytesReadDomainSocketThroughput: string;
  totalBytesReadRemote: string;
  totalBytesReadRemoteThroughput: string;
  totalBytesReadUfs: string;
  totalBytesReadUfsThroughput: string;
  totalBytesWrittenAlluxio: string;
  totalBytesWrittenAlluxioThroughput: string;
  totalBytesWrittenDomainSocket: string;
  totalBytesWrittenDomainSocketThroughput: string;
  totalBytesWrittenUfs: string;
  totalBytesWrittenUfsThroughput: string;
  ufsOps: {
    [key: string]: {
      [key: string]: number;
    };
  },
  ufsReadSize: {
    [key: string]: string;
  },
  ufsWriteSize: {
    [key: string]: string;
  },
  operationMetrics: {
    [key: string]: ICounter;
  }
}

export enum MetricsActionTypes {
  FETCH_REQUEST = '@@metrics/FETCH_REQUEST',
  FETCH_SUCCESS = '@@metrics/FETCH_SUCCESS',
  FETCH_ERROR = '@@metrics/FETCH_ERROR'
}

export interface IMetricsState {
  readonly data: IMetrics;
  readonly errors?: AxiosResponse;
  readonly loading: boolean;
  readonly response?: AxiosResponse;
}
