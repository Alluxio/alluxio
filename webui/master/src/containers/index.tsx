/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export {default as App} from './App/App';

export {default as Browse} from './pages/Browse/Browse';
export {default as Configuration} from './pages/Configuration/Configuration';
export {default as Data} from './pages/Data/Data';
export {default as Overview} from './pages/Overview/Overview';
export {default as MasterLogs} from './pages/MasterLogs/MasterLogs';
export {default as Workers} from './pages/Workers/Workers';
export {default as Metrics} from './pages/Metrics/Metrics';
