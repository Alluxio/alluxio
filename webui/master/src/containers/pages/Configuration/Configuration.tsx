/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';
import {connect} from 'react-redux';
import {Table} from 'reactstrap';
import {compose, Dispatch} from 'redux';

import {withErrors, withLoadingMessage, withFetchData} from '@alluxio/common-ui/src/components';
import {IConfigTriple} from '../../../constants';
import {IApplicationState} from '../../../store';
import {fetchRequest} from '../../../store/config/actions';
import {IConfig} from '../../../store/config/types';
import {IAlertErrors, ICommonState} from "@alluxio/common-ui/src/constants";
import {createAlertErrors} from "@alluxio/common-ui/src/utilities";

interface IPropsFromState extends ICommonState {
  data: IConfig
}

interface IPropsFromDispatch {
  fetchRequest: typeof fetchRequest;
}

export type AllProps = IPropsFromState & IPropsFromDispatch;

export class ConfigurationPresenter extends React.Component<AllProps> {
  public render() {
    const {data} = this.props;

    return (
      <div className="configuration-page">
        <div className="container-fluid">
          <div className="row">
            <div className="col-12">
              <h5>Alluxio Configuration</h5>
              <Table hover={true}>
                <thead>
                <tr>
                  <th>Property</th>
                  <th>Value</th>
                  <th>Source</th>
                </tr>
                </thead>
                <tbody>
                {data.configuration.map((configuration: IConfigTriple) => (
                  <tr key={configuration.left}>
                    <td>
                      <pre className="mb-0"><code>{configuration.left}</code></pre>
                    </td>
                    <td>{configuration.middle}</td>
                    <td>{configuration.right}</td>
                  </tr>
                ))}
                </tbody>
              </Table>
            </div>
            <div className="col-12">
              <h5>Whitelist</h5>
              <Table hover={true}>
                <tbody>
                {data.whitelist.map((whitelist: string) => (
                  <tr key={whitelist}>
                    <td>{whitelist}</td>
                  </tr>
                ))}
                </tbody>
              </Table>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({config, refresh}: IApplicationState): IPropsFromState => {
  const errors: IAlertErrors = createAlertErrors(config.errors !== undefined, []);
  return {
    data: config.data,
    errors: errors,
    loading: config.loading,
    refresh: refresh.data,
    class: 'configuration-page'
  }

};

const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchRequest: () => dispatch(fetchRequest())
});

export default compose(
    connect(mapStateToProps, mapDispatchToProps),
    withFetchData,
    withErrors,
    withLoadingMessage
)(ConfigurationPresenter) as React.Component;
