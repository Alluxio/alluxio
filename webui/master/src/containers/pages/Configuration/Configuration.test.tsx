/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createBrowserHistory, History, LocationState } from 'history';
import React from 'react';
import sinon from 'sinon';

import { AllProps, ConfigurationPresenter } from './Configuration';
import { routePaths } from '../../../constants';
import { createAlertErrors } from '@alluxio/common-ui/src/utilities';
import { IConfig } from '../../../store/config/types';

configure({ adapter: new Adapter() });

describe('Configuration', () => {
  let history: History<LocationState>;
  let props: AllProps;

  beforeAll(() => {
    history = createBrowserHistory({ keyLength: 0 });
    history.push(routePaths.config);

    const testData: IConfig = { configuration: [{ left: 'alluxio.', middle: '', right: '' }], whitelist: ['/'] };

    props = {
      data: testData,
      class: '',
      errors: createAlertErrors(false),
      loading: false,
      fetchRequest: sinon.spy(() => {}),
      refresh: false,
    };
  });

  afterEach(() => {
    sinon.restore();
  });

  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<ConfigurationPresenter {...props} />);
    });

    it('Renders without crashing', () => {
      expect(shallowWrapper.length).toEqual(1);
    });

    it('Matches snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });

    it('Searches data with valid search key', () => {
      shallowWrapper.setState({ searchConfig: 'alluxio.' });
      expect(
        shallowWrapper
          .find('Table')
          .first()
          .dive()
          .find('#filtered-data-body'),
      ).toHaveLength(1);
    });

    it('Searches data with invalid search key', () => {
      const randomStr = Math.random()
        .toString(36)
        .substring(7);
      shallowWrapper.setState({ searchConfig: randomStr });
      expect(
        shallowWrapper
          .find('Table')
          .first()
          .dive()
          .find('#no-data-body'),
      ).toHaveLength(1);
    });
  });
});
