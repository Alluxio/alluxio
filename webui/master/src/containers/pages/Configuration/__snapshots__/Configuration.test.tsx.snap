// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration Shallow component Matches snapshot 1`] = `
<div
  className="configuration-page"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h5>
          Alluxio Configuration
        </h5>
        <Table
          hover={true}
          responsiveTag="div"
          tag="table"
        >
          <thead>
            <tr>
              <th>
                Property
              </th>
              <th>
                Value
              </th>
              <th>
                Source
              </th>
            </tr>
          </thead>
          <tbody />
        </Table>
      </div>
      <div
        className="col-12"
      >
        <h5>
          Whitelist
        </h5>
        <Table
          hover={true}
          responsiveTag="div"
          tag="table"
        >
          <tbody />
        </Table>
      </div>
    </div>
  </div>
</div>
`;
